lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -0,0 +1,145 @@
+from http import HTTPStatus
+from logging import Logger
+
+from flask.json import jsonify
+from flask.wrappers import Request, Response
+from flask_restful import abort
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    ApiBase,
+    API_OPERATION,
+    convert_list,
+    JSON,
+    Parameter,
+    ParamType,
+    Schema,
+    SchemaError,
+)
+from pbench.server.database.models.datasets import (
+    Dataset,
+    DatasetError,
+    DatasetNotFound,
+    Metadata,
+    MetadataError,
+)
+
+
+class DatasetsMetadata(ApiBase):
+    """
+    API class to retrieve and mutate Dataset metadata.
+    """
+
+    GET_KEYWORDS = Parameter(
+        "metadata",
+        ParamType.LIST,
+        element_type=ParamType.KEYWORD,
+        keywords=ApiBase.METADATA,
+    )
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("controller", ParamType.STRING, required=True),
+                Parameter("name", ParamType.STRING, required=True),
+                Parameter(
+                    "metadata",
+                    ParamType.JSON,
+                    keywords=Metadata.USER_UPDATEABLE_METADATA,
+                    required=True,
+                ),
+            ),
+            role=API_OPERATION.UPDATE,
+        )
+
+    def _get(self, _, request: Request) -> Response:
+        """
+        Get the values of client-accessible dataset metadata keys.
+
+        NOTE: This does not rely on a JSON payload to identify the dataset and
+        desired metadata keys. While in theory there's no restriction on
+        passing a request payload to GET, the venerable (obsolete) Javascript
+        requests package doesn't support it, and Elasticsearch allows POST as
+        well as GET for queries because many clients can't support a payload on
+        GET. In this case, we're going to experiment with an alternative, using
+        query parameters.
+
+        GET /api/v1/datasets/metadata?controller=ctrl&name=dname&metadata=SEEN&metadata=SAVED
```

Comment:
`SEEN` and `SAVED` need their `dashboard.` prefixes here, right?  And, at line 103?